src/Orchard.Environment.Extensions/ExtensionManager.cs

```diff
@@ -256,6 +256,11 @@ public Task<ExtensionEntry> LoadExtensionAsync(IExtensionInfo extensionInfo)
 
         public async Task<IEnumerable<FeatureEntry>> LoadFeaturesAsync()
         {
+            // for testing
+            Parallel.ForEach(GetFeatures(),
```

Comment:
@jtkech skype me, I will show you with a Console app what is going on. After that you will understand everything.